src/integrationTest/sam.test.ts

```diff
@@ -103,70 +105,57 @@ async function closeAllEditors(): Promise<void> {
 async function activateExtensions(): Promise<void> {
     console.log('Activating extensions...')
     await activateExtension(VSCODE_EXTENSION_ID.python)
-    await activateExtension(VSCODE_EXTENSION_ID.awstoolkit)
     console.log('Extensions activated')
 }
 
 async function configurePythonExtension(): Promise<void> {
-    logSeparator()
     const configPy = vscode.workspace.getConfiguration('python')
     // Disable linting to silence some of the Python extension's log spam
     await configPy.update('linting.pylintEnabled', false, false)
     await configPy.update('linting.enabled', false, false)
-    logSeparator()
 }
 
 async function configureAwsToolkitExtension(): Promise<void> {
-    logSeparator()
     const configAws = vscode.workspace.getConfiguration('aws')
     await configAws.update('logLevel', 'verbose', false)
     // Prevent the extension from preemptively cancelling a 'sam local' run
-    await configAws.update('samcli.debug.attach.timeout.millis', '90000', false)
-    logSeparator()
+    await configAws.update('samcli.debug.attach.timeout.millis', 90000, false)
 }
 
-function logSeparator() {
-    console.log('************************************************************')
+function runtimeNeedsWorkaround(lang: Language) {
+    return vscode.version.startsWith('1.42') || lang === 'csharp' || lang === 'python'
 }
 
-function configureToolkitLogging() {
-    const logger = getLogger()
-
-    if (logger instanceof WinstonToolkitLogger) {
-        // Ensure we're logging everything possible
-        logger.setLogLevel('debug')
-        // The logs help to diagnose SAM integration test failures
-        logger.logToConsole()
-    } else {
-        assert.fail('Unexpected extension logger')
-    }
-}
-
-describe('SAM Integration Tests', async () => {
+describe('SAM Integration Tests', async function() {
     const samApplicationName = 'testProject'
+    /**
+     * Breadcrumbs from each process, printed at end of all scenarios to give
+     * us an idea of the timeline.
+     */
+    const sessionLog: string[] = []
     let testSuiteRoot: string
 
     before(async function() {
-        // tslint:disable-next-line:no-invalid-this
-        this.timeout(600000)
-
         await activateExtensions()
         await configureAwsToolkitExtension()
         await configurePythonExtension()
 
-        configureToolkitLogging()
-
         testSuiteRoot = await mkdtemp(path.join(projectFolder, 'inttest'))
         console.log('testSuiteRoot: ', testSuiteRoot)
         mkdirpSync(testSuiteRoot)
     })
 
-    after(async () => {
+    after(async function() {
         tryRemoveFolder(testSuiteRoot)
+        // Print a summary of session that were seen by `onDidStartDebugSession`.
+        const sessionReport = `\n    ${sessionLog.join('\n    ')}\n`
```

Comment:
nit: Here's a less repetitive formatting sample. It doesn't have the trailing newline, buyer beware

```suggestion
        const sessionReport = sessionLog.Map(x => `\t{x}`).join('\n')
```

Reply:
will do, though i intentionally avoid tabs, they are less amenable to pasting into comments/discussions, may not render as expected in certain contexts, etc.

Reply:
I'm indifferent on spaces or tabs here. the map+join was the focus of my comment